specification/communication/Communication.Messages/models.tsp

```diff
@@ -0,0 +1,364 @@
+  buttons?: WhatsAppMessageTemplateBindingsButton[];
+}
+
+@doc("The template bindings component for WhatsApp")
+model WhatsAppMessageTemplateBindingsComponent {
+  @doc("The name of the referenced item in the template values.")
+  refValue: string;
+}
+
+@doc("The template bindings component button for WhatsApp")
+model WhatsAppMessageTemplateBindingsButton {
+  @doc("The WhatsApp button sub type")
+  subType: WhatsAppMessageButtonSubType;
+
+  @doc("The name of the referenced item in the template values.")
+  refValue: string;
+}
+
+@doc("Result of the send message operation.")
+model SendMessageResult {
+  @doc("Receipts of the send message operation.")
+  receipts: MessageReceipt[];
+}
+
+@doc("Receipt of the sending one message.")
+model MessageReceipt {
+  @doc("The message id.")
+  messageId: string;
+
+  @doc("The native external platform user identifier of the recipient.")
+  to: string;
+}
+
+@doc("The WhatsApp-specific template response contract")
+model WhatsAppMessageTemplateItem extends MessageTemplateItem {
+  #suppress "@azure-tools/typespec-azure-core/no-unknown" "Payload being returned from WhatsApp with no guaranteed object type."
+  @doc("WhatsApp platform's template content. This is the payload returned from WhatsApp API.")
+  content?: unknown;
+
+  @doc("Message template response type is whatsApp.")
+  kind: CommunicationMessagesChannel.whatsApp;
+}
+
+@resource("messages/notifications")
+@doc("A notification.")
+model Notifications {
+  @key
+  @visibility("read")
+  @doc("The notification ID.")
+  id: string;
+}
+
+@resource("messages/channels")
+@doc("A communication channel.")
+model CommunicationChannel {
+  @key
+  @visibility("read")
+  @doc("The registration ID of the channel.")
+  channelId: string;
+
+  @doc("Number of objects to return per page.")
+  @query
+  @visibility("create")
+  maxpagesize?: int32;
+}
+
+@resource("messages/streams")
+@doc("A data stream.")
+model MessageDataStream {
+  @key
+  @visibility("read")
+  @doc("The stream ID.")
+  id: string;
+
+  @body
+  @doc("The stream body.")
+  body: bytes;
+
+  @header
+  @doc("The stream content type.")
+  contentType: "application/octet-stream";
+}
+
+@resource("templates")
+@parentResource(CommunicationChannel)
+@discriminator("kind")
+@doc("The message template as returned from the service.")
+model MessageTemplateItem {
+  @key
+  @visibility("read")
+  @doc("The template's name.")
+  name: string;
+
+  @doc("The template's language, in the ISO 639 format, consist of a two-letter language code followed by an optional two-letter country code, e.g., 'en' or 'en_US'.")
+  language: string;
+
+  @doc("The aggregated template status.")
+  status: MessageTemplateStatus;
+
+  @doc("The type discriminator describing a template type.")
+  kind: CommunicationMessagesChannel;
+}
+
+alias ListTemplatesQueryParams = ListQueryParametersTrait<{
+  @doc("Number of objects to return per page.")
+  @query
+  maxpagesize?: int32;
```

Comment:
@glorialimicrosoft Can we try using the https://azure.github.io/typespec-azure/docs/libraries/azure-core/reference/data-types#Azure.Core.MaxPageSizeQueryParameter type instead of defining your own?

Reply:
Actually this is not needed here, ListTemplatesQueryParams already takes care of it. Removed.